tracer/src/Datadog.Trace.Tools.dd_dotnet/CreatedumpCommand.cs

```diff
@@ -0,0 +1,676 @@
+
+        var function = (delegate* unmanaged<int, IntPtr>)export;
+
+        var ptr = function(pid);
+
+        if (ptr == IntPtr.Zero)
+        {
+            Errors.Add("Failed to create crash report");
+            return;
+        }
+
+        using var iunknown = NativeObjects.IUnknown.Wrap(ptr);
+
+        int result = iunknown.QueryInterface(ICrashReport.Guid, out var crashReportPtr);
+
+        if (result != 0)
+        {
+            Errors.Add($"Failed to query interface: {result}");
+            return;
+        }
+
+        using var crashReport = NativeObjects.ICrashReport.Wrap(crashReportPtr);
+
+        try
+        {
+            crashReport.Initialize();
+        }
+        catch (Win32Exception ex)
+        {
+            Errors.Add($"Failed to initialize crash report: {GetLastError(crashReport, ex)}");
+            return;
+        }
+
+        if (crashThread == null)
+        {
+            var firstThreadWithException = _runtime.Threads.FirstOrDefault(t => t.CurrentException != null);
+
+            if (firstThreadWithException != null)
+            {
+                crashThread = (int)firstThreadWithException.OSThreadId;
+            }
+        }
+
+        ClrException? exception = null;
+
+        // Check if there's an exception on the crash thread
+        if (crashThread != null)
+        {
+            exception = _runtime.Threads.FirstOrDefault(t => t.OSThreadId == crashThread.Value)?.CurrentException;
+        }
+
+        bool isSuspicious;
+
+        try
+        {
+            var callback = (delegate* unmanaged<IntPtr, ResolveMethodData*, int>)&ResolveManagedMethod;
+            crashReport.ResolveStacks(crashThread ?? 0, (IntPtr)callback, out isSuspicious);
+        }
+        catch (Win32Exception ex)
+        {
+            Errors.Add($"Failed to resolve stacks: {GetLastError(crashReport, ex)}");
+            return;
+        }
+
+        if (!isSuspicious && exception != null)
+        {
+            // The stacks aren't suspicious, but maybe the exception is
+            var exceptionType = exception.Type?.Name ?? string.Empty;
+
+            var suspiciousExceptionTypes = new[] { "System.InvalidProgramException", "System.Security.VerificationException", "System.MissingMethodException", "System.BadImageFormatException" };
+
+            if (exceptionType.StartsWith("Datadog") || suspiciousExceptionTypes.Contains(exceptionType))
+            {
+                isSuspicious = true;
+            }
+        }
+
+        if (!isSuspicious)
+        {
+            return;
+        }
+
+        AnsiConsole.WriteLine("Datadog - The crash might have been caused by automatic instrumentation, sending crash report...");
+
+        if (signal.HasValue)
+        {
+            _ = SetSignal(crashReport, signal.Value);
+        }
+
+        _ = SetMetadata(crashReport, _runtime, exception);
+
+        try
+        {
+            var outputFile = Environment.GetEnvironmentVariable("DD_TRACE_CRASH_OUTPUT");
```

Comment:
Do we want this to _prevent_ sending, or to allow _additional_ output to a file? 🤔

Reply:
It is really intended for tests, so I think it's better if it prevents sending as well.